Makefile

```diff
@@ -12,21 +12,24 @@ IO_TEST_DIR		:= io_test
 UNIT_TEST_DIR	:= unit_test
 JSON_FILE_DIR	:= json_files
 
+CPPFLAGS := -MMD -MP
 CXXFLAGS := -std=c++17 -Wall -g
 
-.PHONY: all clean
+.PHONY: all clean documentation io-test unit-test static-code-analysis memory-leak-test
 
 all: $(EXE)
 
 $(EXE): $(OBJ)
 	$(CXX) $^ -o $@
 
 $(OBJ_DIR)/%.o: $(SRC_DIR)/%.cpp | $(OBJ_DIR)
-	$(CXX) $(CXXFLAGS) -c $< -o $@
+	$(CXX) $(CPPFLAGS) $(CXXFLAGS) -c $< -o $@
```

Comment:
Tovabbra is buta vagyok ehhez. Azt ertem, hoy a -MP legeneralja a make targetet. a -c utani magic van az en tudasszintem felett. Nem latom, hogy ezek a targetek hogy kerulnek bele a makefile-ba. @smidlaj, ra tudnal nezni?

Reply:
A % os rész kiszedi a .cpp hez tartozó .o fájlt a szabály szerint, a $< az target első előfeltételét adja vissza, a $@ pedig magát a targetet.

Reply:
Koszi, ma is tanultam valamit. De azt meg mindig nem latom ebbol, hogy a header dependency-k hogy kerulnek be.

Reply:
A -MMD legenerálja a .d dependency listákat és a makefileban hozzá vannak adva a -include $(OBJ:.o=.d) résznél.

Reply:
Na ezt a reszt nem vettem eszre, koszi. Mar csak egy dolgot nem ertek: Ez a recept generalna le azokat a .d fajlokat, de ha nem lat valtozast a cpp-ben, akkor nem fut le a g++, nem generalodik a .d es ebbol nem fogja tudni, melyik headert rantsa be. Nem kellenenek valahol a .d targetek is?